GiniVision/Classes/CameraViewController.swift

```diff
@@ -103,23 +120,36 @@ public typealias CameraErrorBlock = (_ error: CameraError) -> ()
     }
     
     // Output
-    fileprivate var successBlock: CameraSuccessBlock?
-    fileprivate var errorBlock: CameraErrorBlock?
+    fileprivate var successBlock: CameraScreenSuccessBlock?
+    fileprivate var failureBlock: CameraScreenFailureBlock?
     
     /**
-     Designated intitializer for the `CameraViewController` which allows to set a success block and an error block which will be executed accordingly.
+     Designated initializer for the `CameraViewController` which allows to set a success block and an error block which will be executed accordingly.
      
-     - parameter success: Success block to be executed when image was taken.
+     - parameter success: Success block to be executed when document was picked or image was taken.
      - parameter failure: Error block to be executed if an error occurred.
      
-     - returns: A view controller instance allowing the user to take a picture.
+     - returns: A view controller instance allowing the user to take a picture or pick a document.
      */
-    public init(success: @escaping CameraSuccessBlock, failure: @escaping CameraErrorBlock) {
+    public init(successBlock: @escaping CameraScreenSuccessBlock, failureBlock: @escaping CameraScreenFailureBlock) {
+        filePickerManager = FilePickerManager()
         super.init(nibName: nil, bundle: nil)
         
         // Set callback
-        successBlock = success
-        errorBlock = failure
+        self.successBlock = successBlock
+        self.failureBlock = failureBlock
+        
+        // Configure file picker
+        filePickerManager.didPickFile = { [unowned self] document in
+            do {
+                try document.validate()
```

Comment:
Just curious.. why did you choose to use exceptions rather than returning errors here?

Reply:
To be consistent with the other source configuration (camera).